### PR TITLE
[ja] Translate tasks/debug-application-cluster/local-debugging into Japanese

### LINKED ISSUE #30877: [ja] Translate tasks/debug-application-cluster/local-debugging into Japanese
# This is a Feature Request

## What would you like to be added

Translate [tasks/debug-application-cluster/local-debugging](https://kubernetes.io/docs/tasks/debug-application-cluster/local-debugging/)  into Japanese.

## Why is this needed

This page has not been translated.

## Comments

/language ja
/assign

### PR DESCRIPTION
**Pull request by ptux**
resolve #30877

### CODE REVIEW

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: ローカルでのサービス開発・デバッグ
+reviewers:
+- ptux
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:4]: reviewersブロックは削除して下さい

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+content_type: task
+---
+
+<!-- overview -->
+
+Kubernetesアプリケーションは通常、複数の独立したサービスから構成され、それぞれが独自のコンテナで動作しています。これらのサービスをリモートのKubernetesクラスタ上で開発・デバッグするには、[get a shell on a running container](/docs/task/debug-application-cluster/get-shell-running-container/)してリモートシェル内でツールを実行しなければならず面倒な場合があります。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:10]:
```suggestion
Kubernetesアプリケーションは通常、複数の独立したサービスから構成され、それぞれが独自のコンテナで動作しています。これらのサービスをリモートのKubernetesクラスター上で開発・デバッグするには、[get a shell on a running container](/docs/task/debug-application-cluster/get-shell-running-container/)してリモートシェル内でツールを実行しなければならず面倒な場合があります。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+`telepresence`は、リモートKubernetesクラスタにサービスをプロキシングしながら、ローカルでサービスを開発・デバッグするプロセスを容易にするためのツールです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:12]:
```suggestion
`telepresence`は、リモートKubernetesクラスターにサービスをプロキシングしながら、ローカルでサービスを開発・デバッグするプロセスを容易にするためのツールです。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+`telepresence` を使用すると、デバッガや IDE などのカスタムツールをローカルサービスで使用でき、ConfigMap や secret、リモートクラスタ上で動作しているサービスへのフルアクセスをサービスに提供します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:13]:
```suggestion
`telepresence` を使用すると、デバッガーやIDEなどのカスタムツールをローカルサービスで使用でき、ConfigMapやsecret、リモートクラスター上で動作しているサービスへのフルアクセスをサービスに提供します。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+このドキュメントでは、リモートクラスタ上で動作しているサービスをローカルで開発・デバッグするために `telepresence` を使用する方法を説明します。
+
+## {{% heading "prerequisites" %}}
+
+* Kubernetesクラスタがインストールされていること
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:19]:
```suggestion
* Kubernetesクラスターがインストールされていること
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+* クラスタと通信するために `kubectl` が設定されていること
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:20]:
```suggestion
* クラスターと通信するために `kubectl` が設定されていること
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+* telepresence(https://www.telepresence.io/reference/install)がインストールされていること
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:21]:
```suggestion
* [telepresence](https://www.telepresence.io/reference/install)がインストールされていること
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+<!-- steps -->
+
+## リモートクラスタ上でシェルの取得
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:25]:
```suggestion
## リモートクラスター上でシェルの取得
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+ターミナルを開いて、引数なしで`telepresence`を実行すると、`telepresence`シェルが表示されます。
+このシェルはローカルで動作し、ローカルのファイルシステムに完全にアクセスすることができます。
+
+この`telepresence`シェルは様々な方法で使用することができます。
+例えば、ラップトップでシェルスクリプトを書いて、それをシェルから直接リアルタイムで実行することができます。これはリモートシェルでもできますが、好みのコードエディタが使えないかもしれませんし、コンテナが終了するとスクリプトは削除されます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:31]:
```suggestion
例えば、ラップトップでシェルスクリプトを書いて、それをシェルから直接リアルタイムで実行することができます。これはリモートシェルでもできますが、好みのコードエディターが使えないかもしれませんし、コンテナが終了するとスクリプトは削除されます。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+終了してシェルを閉じるには`exit`と入力してください。
+
+## 既存サービスの開発・デバッグ
+
+When developing an application on Kubernetes, you typically program or debug a single service. The service might require access to other services for testing and debugging. One option is to use the continuous deployment pipeline, but even the fastest deployment pipeline introduces a delay in the program or debug cycle.
+
+Use the `--swap-deployment` option to swap an existing deployment with the Telepresence proxy. Swapping allows you to run a service locally and connect to the remote Kubernetes cluster. The services in the remote cluster can now access the locally running instance.
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:39]: これらのブロックは消し忘れだと思います

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+
+Kubernetes上でアプリケーションを開発する場合、通常は1つのサービスをプログラミングまたはデバッグすることになります。
+そのサービスは、テストやデバッグのために他のサービスへのアクセスを必要とする場合があります。
+継続的なデプロイメントパイプラインを使用することも一つの選択肢ですが、最速のデプロイメントパイプラインでさえ、プログラムやデバッグサイクルに遅延が発生します。
+
+既存のデプロイメントとtelepresenceプロキシを交換するには、`--swap-deployment` オプションを使用します。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:45]:
```suggestion
既存のデプロイメントとtelepresenceプロキシーを交換するには、`--swap-deployment` オプションを使用します。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+スワップすることで、ローカルでサービスを実行し、リモートのKubernetesクラスタに接続することができます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:46]:
```suggestion
スワップすることで、ローカルでサービスを実行し、リモートのKubernetesクラスターに接続することができます。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 13]
```diff
@@ -0,0 +1,67 @@
...
+リモートクラスタ内のサービスは、ローカルで実行されているインスタンスにアクセスできるようになりました。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:47]:
```suggestion
リモートクラスター内のサービスは、ローカルで実行されているインスタンスにアクセスできるようになりました。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+telepresenceを「--swap-deployment」で実行するには、次のように入力します。
+
+`telepresence --swap-deployment $DEPLOYMENT_NAME`
+
+ここで、$DEPLOYMENT_NAMEは既存のディプロイメントの名前です。
+
+このコマンドを実行すると、シェルが起動します。そのシェルで、サービスを起動します。
+そして、ローカルでソースコードの編集を行い、保存すると、すぐに変更が反映されるのを確認できます。
+また、デバッガやその他のローカルな開発ツールでサービスを実行することもできます。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:57]:
```suggestion
また、デバッガーやその他のローカルな開発ツールでサービスを実行することもできます。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]
I'm verry sorry to be late to review, however the base page has big update 10 days ago...

/cc @kakts @bells17 @makocchi-git

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 15]
```diff
@@ -0,0 +1,67 @@
...
+
+## {{% heading "whatsnext" %}}
+
+もしハンズオンのチュートリアルに興味があるなら、Google Kubernetes Engine上でGuestbookアプリケーションをローカルに開発する手順を説明した[this tutorial](https://cloud.google.com/community/tutorials/developing-services-with-k8s)をチェックしてみてください。
+
+telepresenceには、状況に応じて[numerous proxying options](https://www.telepresence.io/reference/methods)があります。
+
+さらに詳しい情報は、[telepresence website](https://www.telepresence.io)をご覧ください。
+
+
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:67]:
```suggestion
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 16]
```diff
@@ -0,0 +1,67 @@
...
+---
+
+<!-- overview -->
+
+Kubernetesアプリケーションは通常、複数の独立したサービスから構成され、それぞれが独自のコンテナで動作しています。これらのサービスをリモートのKubernetesクラスタ上で開発・デバッグするには、[get a shell on a running container](/docs/task/debug-application-cluster/get-shell-running-container/)してリモートシェル内でツールを実行しなければならず面倒な場合があります。
+
+`telepresence`は、リモートKubernetesクラスタにサービスをプロキシングしながら、ローカルでサービスを開発・デバッグするプロセスを容易にするためのツールです。
+`telepresence` を使用すると、デバッガや IDE などのカスタムツールをローカルサービスで使用でき、ConfigMap や secret、リモートクラスタ上で動作しているサービスへのフルアクセスをサービスに提供します。
+
+このドキュメントでは、リモートクラスタ上で動作しているサービスをローカルで開発・デバッグするために `telepresence` を使用する方法を説明します。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:15]:
```suggestion
このドキュメントでは、リモートクラスタ上で動作しているサービスをローカルで開発・デバッグするために`telepresence`を使用する方法を説明します。
```

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 17]
```diff
@@ -0,0 +1,67 @@
...
+
+ここで、$DEPLOYMENT_NAMEは既存のディプロイメントの名前です。
+
+このコマンドを実行すると、シェルが起動します。そのシェルで、サービスを起動します。
+そして、ローカルでソースコードの編集を行い、保存すると、すぐに変更が反映されるのを確認できます。
+また、デバッガやその他のローカルな開発ツールでサービスを実行することもできます。
+
+## {{% heading "whatsnext" %}}
+
+もしハンズオンのチュートリアルに興味があるなら、Google Kubernetes Engine上でGuestbookアプリケーションをローカルに開発する手順を説明した[this tutorial](https://cloud.google.com/community/tutorials/developing-services-with-k8s)をチェックしてみてください。
```
Comment (riita10069) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:61]:
```suggestion
もしハンズオンのチュートリアルに興味があるなら、Google Kubernetes Engine上でGuestbookアプリケーションをローカルに開発する手順を説明した[こちらのチュートリアル](https://cloud.google.com/community/tutorials/developing-services-with-k8s)をチェックしてみてください。
```

**Comment by nasa9084** [comment]
see folded comments too

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 18]
```diff
@@ -0,0 +1,60 @@
+---
+title: ローカルでのサービス開発・デバッグ
+content_type: task
+---
+
+<!-- overview -->
+
+Kubernetesアプリケーションは通常、複数の独立したサービスから構成され、それぞれが独自のコンテナで動作しています。これらのサービスをリモートのKubernetesクラスター上で開発・デバッグするには、[get a shell on a running container](/docs/task/debug-application-cluster/get-shell-running-container/)してリモートシェル内でツールを実行しなければならず面倒な場合があります。
+
+`telepresence`は、リモートKubernetesクラスターにサービスをプロキシングしながら、ローカルでサービスを開発・デバッグするプロセスを容易にするためのツールです。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:10]:
```suggestion
`telepresence`は、リモートKubernetesクラスターにサービスをプロキシーしながら、ローカルでサービスを開発・デバッグするプロセスを容易にするためのツールです。
```

**Review by nasa9084 — CHANGES_REQUESTED** [review]

File: content/ja/docs/tasks/debug-application-cluster/local-debugging.md [hunk 19]
```diff
@@ -0,0 +1,60 @@
...
+
+既存のデプロイメントとtelepresenceプロキシーを交換するには、`--swap-deployment` オプションを使用します。
+スワップすることで、ローカルでサービスを実行し、リモートのKubernetesクラスターに接続することができます。
+リモートクラスター内のサービスは、ローカルで実行されているインスタンスにアクセスできるようになりました。
+
+telepresenceを「--swap-deployment」で実行するには、次のように入力します。
+
+`telepresence --swap-deployment $DEPLOYMENT_NAME`
+
+ここで、$DEPLOYMENT_NAMEは既存のディプロイメントの名前です。
```
Comment (nasa9084) [inline content/ja/docs/tasks/debug-application-cluster/local-debugging.md:47]:
```suggestion
ここで、$DEPLOYMENT_NAMEは既存のDeploymentの名前です。
```

**Comment by kakts** [comment]
Thank you for your contribution.

/lgtm

**Pull request merged**